worker/worker/parser/category.py

```diff
@@ -16,97 +18,152 @@ def extract_challenge_ids(txt):
+    return description1, description2
+
+
+def extract_category_prereq(r):
+    tree = html.fromstring(r.content)
+    result = tree.xpath('string(//div[starts-with(@class, "texte crayon rubrique-texte")]/p[starts-with(., '
+                        '"Préreq")]/following-sibling::p)')  # if prerequisites are on two "p" html tags
     if not result:
-        raise RootMeParsingError("Could not parse description.")
+        result = tree.xpath('string(//div[starts-with(@class, "texte crayon rubrique-texte")]/p[starts-with(., '
+                            '"Préreq")])')
+    result = list(filter(lambda x: '\n' not in x, result.split('\xa0')[1:]))
+    return result
 
-    description1 = result[0]
 
-    pattern = '<div class="texte crayon rubrique-texte.*?><p>(.*?)</p>'
-    result = re.findall(pattern, txt)
-    description2 = result[0] if result else ''
+def extract_paths(r):
+    tree = html.fromstring(r.content)
+    paths = tree.xpath('//td[@class="text-left"]/a[starts-with(@href, "fr/Challenges")]/@href')
+    return paths
 
-    return unescape(description1), unescape(description2)
 
+def extract_statements(r):
+    tree = html.fromstring(r.content)
+    statements = tree.xpath('//td[@class="text-left"]/a[starts-with(@href, "fr/Challenges")]/@title')
+    return statements
 
-def extract_category_prereq(txt):
-    txt = txt.replace('\n', '')
-    txt = txt.replace('&nbsp;', '')
 
-    pattern = '<p>Prérequis\xa0:(.*?)\/p>'
-    result = re.findall(pattern, txt)
+def extract_names(r):
+    tree = html.fromstring(r.content)
+    names = tree.xpath('//td[@class="text-left"]/a[starts-with(@href, "fr/Challenges")]/text()')
+    return names
 
-    if result == ['<']:
-        pattern = '<p>Prérequis\xa0:(.*?)\/div>'
-        result = re.findall(pattern, txt)
 
-    if not result:
-        return []
+def extract_validations_percentages(r):
+    tree = html.fromstring(r.content)
+    validations_percentages = tree.xpath('//span[starts-with(@class, "gras left text-left")]/text()')
+    return validations_percentages
 
-    pattern = '><img.*?>\xa0(.*?)<'
-    result = re.findall(pattern, result[0])
-    if not result:
-        raise RootMeParsingError("Could not parse prerequisites.")
 
-    return [r.replace('\xa0', '') for r in result]
+def extract_validations_nbs(r):
+    tree = html.fromstring(r.content)
+    validations_nbs = tree.xpath('//span[@class="right"]/a/text()')
+    return validations_nbs
 
 
-def extract_challenge_rows(txt):
-    txt = txt.replace('\n', '')
-    txt = txt.replace('&nbsp;', '')
-    pattern = '<tr>(.*?)</tr>'
-    result = re.findall(pattern, txt)
-    if not result:
-        raise RootMeParsingError("Could not parse challenges.")
-    return result
+def extract_difficulties(r):
+    tree = html.fromstring(r.content)
+    difficulties = tree.xpath('//td[@class="show-for-medium-up"]/a[starts-with(@href,"tag")]/@title')
+    difficulties = list(map(lambda x: x.split(':')[0].strip(), difficulties))
```

Comment:
`[x.split(':')[0] for x in difficulties]`

Reply:
OK